api/environments/identities/traits/models.py

```diff
@@ -3,14 +3,31 @@
 from django.core.exceptions import ObjectDoesNotExist
 from django.db import models
 from django.utils.encoding import python_2_unicode_compatible
-from simple_history.models import HistoricalRecords
+from django_lifecycle import AFTER_DELETE, AFTER_SAVE, LifecycleModel, hook
 
 from environments.identities.traits.exceptions import TraitPersistenceError
 from environments.models import BOOLEAN, FLOAT, INTEGER, STRING
 
 
+class TriatQuerySet(models.QuerySet):
+    def delete(self, *args, **kwargs):
+        # collect all the identities that we need to update
+        identitie_to_delete = [trait.identity for trait in self]
```

Comment:
```suggestion
        identities_to_delete = {trait.identity for trait in self}
```

Although isn't this `identities_to_update` since we're not actually deleting them from dynamo, right? We're just updating them to remove some traits?